Tools/clinic/clinic.py

```diff
@@ -4602,7 +4608,10 @@ def parse(self, block: Block) -> None:
         for line_number, line in enumerate(lines, self.clinic.block_parser.block_start_line_number):
             if '\t' in line:
                 fail('Tab characters are illegal in the Clinic DSL.\n\t' + repr(line), line_number=block_start)
-            self.state(line)
+            try:
+                self.state(line)
+            except ClinicError as exc:
+                raise ClinicError(str(exc), filename=exc.filename, lineno=line_number)
```

Comment:
Thoughts on mutating the existing exception and re-raising it, rather than raising a new exception?

```suggestion
            try:
                self.state(line)
            except ClinicError as exc:
                exc.lineno = line_number
                raise
```

Reply:
Thanks, I didn't think of that!